Add controllers

BorodaPT opened the pull request.

BorodaPT committed:
refactor:
1. Для сущностей Item и User созданы наборы классов:
- Controller(обработка http запросов)
- Service
- Repository(хранение в памяти данных)
- DTO
- Mapper
2. Добавлены поля сущностей Booking и ItemRequest

BorodaPT committed:
code-stile

BorodaPT committed:
code-stile:

db-exp commented on src/main/java/ru/practicum/shareit/item/ItemController.java:
```diff
@@ -1,12 +1,61 @@
+    public List<ItemDto> search(@RequestParam(name = "text", required = false) String text) {
+        return itemService.search(text).stream().map(itemMapper::toDTO).collect(toList());
+    }
+
```
лишние строчки

db-exp commented on src/main/java/ru/practicum/shareit/item/dto/ItemDto.java:
```diff
@@ -1,7 +1,36 @@
+        this.name = name;
+        this.description = description;
+        this.isAvailable = isAvailable;
+        //this.owner = owner;
```
ненужный код

db-exp commented on src/main/java/ru/practicum/shareit/user/UserMapper.java:
```diff
@@ -0,0 +1,22 @@
+
+@AllArgsConstructor
+@Component
+public class UserMapper {
```
лучше не делать компонентом, а сделать методы статическими

db-exp commented on src/main/java/ru/practicum/shareit/item/dto/ItemDto.java:
```diff
@@ -1,7 +1,36 @@
+    private String name;
+
+    @NotEmpty
+    private String description;
```
можно еще аннотацию Size добавить

db-exp commented on src/main/java/ru/practicum/shareit/item/dto/ItemDto.java:
```diff
@@ -1,7 +1,36 @@
+    private final Long id;
+
+    @NotEmpty
+    private String name;
```
можно еще аннотацию Size добавить

db-exp commented on src/main/java/ru/practicum/shareit/item/ItemRepositoryInMemory.java:
```diff
@@ -0,0 +1,100 @@
+        if (items.containsKey(idItem)) {
+            Item itemBase = items.get(idItem);
+            if (itemBase.getOwner() != idUser) {
+                throw new ExceptionNotFound("updateItem","Операция вызвана не владельцем");
```
тут логичней ForbiddenException или как-то так

BorodaPT committed:
code-stile:
- удалены пустые строки
- удалены заакомментированые строки
refactor:
- мапперы Item и User сделаны статичными
- добавлены ограничения на длину полей ItemDto.description ItemDto.name
- Доработан ExceptionDataRequest для более гибкой установки статусов response
